app/controllers/auth/registrations_controller.rb

```diff
@@ -126,4 +129,16 @@ def require_not_suspended!
   def set_cache_headers
     response.headers['Cache-Control'] = 'no-cache, no-store, max-age=0, must-revalidate'
   end
+
+  def check_registration_form_time!
```

Comment:
Maybe this could be moved to a validator?

Reply:
I'm not sure, it would require moving the logic down to the model, wouldn't it? Also, I don't think `simple_forms` handles displaying validation errors well when they aren't about an attribute it generates an input for.

Reply:
It would require you to pass the form rendering time into the model as a virtual attribute, but it would be a lot cleaner than this method. Also, we render the errors in our own partial, so if there is missing support for errors without an attribute, it can be easily added

Reply:
https://github.com/tootsuite/mastodon/blob/master/app/views/shared/_error_messages.html.haml

Reply:
Moved to the User model validation